saleor/graphql/payment/mutations.py

```diff
@@ -824,16 +833,22 @@ def create_transaction_event(
         )
 
     @classmethod
-    def add_amounts_to_order(cls, order_id: uuid.UUID, transaction_data: dict):
-        authorized_amount = transaction_data.get("authorized_value", 0)
-        charged_amount = transaction_data.get("charged_value", 0)
-        if not authorized_amount and not charged_amount:
-            return
-        add_to_order_total_authorized_and_total_charged(
-            order_id=order_id,
-            authorized_amount_to_add=authorized_amount,
-            charged_amount_to_add=charged_amount,
-        )
+    def update_order(cls, order: order_models.Order, transaction_data: dict):
+        order_id = transaction_data["order_id"]
+        authorized_amount = transaction_data.get("authorized_value", Decimal(0))
+        charged_amount = transaction_data.get("charged_value", Decimal(0))
+        updated_fields = {}
+        if authorized_amount or charged_amount:
+            updated_fields = order_total_authorized_and_total_charged(
+                authorized_amount_to_add=authorized_amount,
+                charged_amount_to_add=charged_amount,
+            )
+        if (
+            order.channel.automatically_confirm_all_new_orders
```

Comment:
Have you considered extending `update_order_status` to handle such scenario? I think that there is multiple places where we use channel to calculate the order status, so I think maybe it would be worth to add it there?

Reply:
Yeah, actually I had that in mind, but forget to change that later. Will do that.